### PR TITLE
Turn off built-in .NET SDK analyzers if user installs analyzer FxCopAnalyzers or NetAnalyzers NuGet package

### PR DESCRIPTION
**Pull request by mavasani**
~~**Review note:** _Changes to .md and .sarif files in this PR are auto-generated by post-build tooling that runs on pack for the solution. They can be ignored for review. The only files requiring review are the first file (`eng\GenerateAnalyzerNuspec.targets`) and last file (`src/Tools/GenerateDocumentationAndConfigFiles/Program.cs`) in this PR._~~ No longer applicable.

Starting .NET5, we insert the analyzer assemblies with all the CA analyzers from `Microsoft.CodeAnalysis.NetAnalyzers` NuGet package. These analyzers are enabled by default for net5 projects or can be enabled for any .NET project by setting MSBuild property `EnableNETAnalyzers`. However, this leads to duplicate analyzers if users explicitly install `Microsoft.CodeAnalysis.FxCopAnalyzers` NuGet package or `Microsoft.CodeAnalysis.NetAnalyzers` NuGet package. This PR drops in additional props file in the above NuGet packages that will explicitly set `EnableNETAnalyzers = false`. This additional props file should only be present in the NuGet packages, and not inserted into the .NET SDK, to get the correct de-dupe behavior without affecting defaults.

**Testing:**
1. Created a .net5 project using latest .NET SDK with a NuGet package reference to FxCop analyzers `3.3.0-beta2.final`:

```xml
  <PropertyGroup>
    <TargetFramework>net5</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0-beta2.final">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>
```

2. Verified duplicate CA diagnostics and analyzer assemblies in Analyzers node

image

3. Upgraded the project to `3.3.0-dev` package with packages built locally from this branch

4. Verified the built-in .NET SDK analyzers are disabled and no duplicate CA diagnostics or analyzer assemblies in Analyzers node.

image

5. Repeated the same exercise as above 4 steps with Microsoft.CodeAnalysis.NetAnalyzers instead of Microsoft.CodeAnalysis.FxCopAnalyzers and confirmed the same duplicates on beta2 analyzer packages, and fixes with locally built packages

**Before:**
image

**After:**
image

### CODE REVIEW

File: src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md [hunk 1]
```diff
@@ -2,15 +2,12 @@
...
-CA1831 | https://docs.microsoft.com/visualstudio/code-quality/ca1831 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1832 | https://docs.microsoft.com/visualstudio/code-quality/ca1832 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1833 | https://docs.microsoft.com/visualstudio/code-quality/ca1833 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1834 | https://docs.microsoft.com/visualstudio/code-quality/ca1834 | Consider using 'StringBuilder.Append(char)' when applicable. |
-CA1835 | https://docs.microsoft.com/visualstudio/code-quality/ca1835 | Prefer the 'Memory'-based overloads for 'ReadAsync' and 'WriteAsync'. |
-CA1836 | https://docs.microsoft.com/visualstudio/code-quality/ca1836 | Prefer IsEmpty over Count |
+CA1834 | https://docs.microsoft.com/visualstudio/code-quality/ca1834 | Consider using 'StringBuilder.Append(char)' when applicable |
+CA1837 | https://docs.microsoft.com/visualstudio/code-quality/ca1837 | Use 'Environment.ProcessId' |
+CA1838 | https://docs.microsoft.com/visualstudio/code-quality/ca1838 | Avoid 'StringBuilder' parameters for P/Invokes |
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:7]: FYI elinor-fung - the analyzers you recently added need to be documented as per the Documentation requirements at https://github.com/dotnet/roslyn-analyzers/blob/master/GuidelinesForNewRules.md

File: src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md [hunk 2]
```diff
@@ -2,15 +2,12 @@
...
 Rule ID | Missing Help Link | Title |
 --------|-------------------|-------|
-CA1830 | https://docs.microsoft.com/visualstudio/code-quality/ca1830 | Prefer strongly-typed Append and Insert method overloads on StringBuilder. |
-CA1831 | https://docs.microsoft.com/visualstudio/code-quality/ca1831 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1832 | https://docs.microsoft.com/visualstudio/code-quality/ca1832 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1833 | https://docs.microsoft.com/visualstudio/code-quality/ca1833 | Use AsSpan or AsMemory instead of Range-based indexers when appropriate |
-CA1834 | https://docs.microsoft.com/visualstudio/code-quality/ca1834 | Consider using 'StringBuilder.Append(char)' when applicable. |
-CA1835 | https://docs.microsoft.com/visualstudio/code-quality/ca1835 | Prefer the 'Memory'-based overloads for 'ReadAsync' and 'WriteAsync'. |
-CA1836 | https://docs.microsoft.com/visualstudio/code-quality/ca1836 | Prefer IsEmpty over Count |
+CA1834 | https://docs.microsoft.com/visualstudio/code-quality/ca1834 | Consider using 'StringBuilder.Append(char)' when applicable |
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:5]: @pgovind Do you plan to add documentation for this rule?

File: src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md [hunk 3]
```diff
@@ -2,15 +2,12 @@
...
+CA1837 | https://docs.microsoft.com/visualstudio/code-quality/ca1837 | Use 'Environment.ProcessId' |
+CA1838 | https://docs.microsoft.com/visualstudio/code-quality/ca1838 | Avoid 'StringBuilder' parameters for P/Invokes |
 CA2008 | https://docs.microsoft.com/visualstudio/code-quality/ca2008 | Do not create tasks without passing a TaskScheduler |
-CA2012 | https://docs.microsoft.com/visualstudio/code-quality/ca2012 | Use ValueTasks correctly |
-CA2014 | https://docs.microsoft.com/visualstudio/code-quality/ca2014 | Do not use stackalloc in loops. |
-CA2247 | https://docs.microsoft.com/visualstudio/code-quality/ca2247 | Argument passed to TaskCompletionSource constructor should be TaskCreationOptions enum instead of TaskContinuationOptions enum. |
 CA2249 | https://docs.microsoft.com/visualstudio/code-quality/ca2249 | Consider using 'string.Contains' instead of 'string.IndexOf' |
+CA2361 | https://docs.microsoft.com/visualstudio/code-quality/ca2361 | Ensure autogenerated class containing DataSet.ReadXml() is not used with untrusted data |
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:10]: FYI @dotpaul - these security rules need documentation

File: src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:7]: Tagging @elinor-fung for real

Comment (elinor-fung) [reply src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:7]: Yep - doc for CA1838 is in PR: https://github.com/MicrosoftDocs/visualstudio-docs-pr/pull/6959

Comment (mavasani) [reply src/Microsoft.NetCore.Analyzers/RulesMissingDocumentation.md:7]: THanks!

**Comment by mavasani** [comment]
Thanks @jmarolf!

**Pull request merged**